codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -141,18 +141,6 @@ public void close() {
         frameReader.close();
     }
 
-    /**
-     * Calculate the threshold in bytes which should trigger a {@code GO_AWAY} if a set of headers exceeds this amount.
-     * @param maxHeaderListSize
-     *      <a href="https://tools.ietf.org/html/rfc7540#section-6.5.2">SETTINGS_MAX_HEADER_LIST_SIZE</a> for the local
-     *      endpoint.
-     * @return the threshold in bytes which should trigger a {@code GO_AWAY} if a set of headers exceeds this amount.
-     */
-    protected long calculateMaxHeaderListSizeGoAway(long maxHeaderListSize) {
```

Comment:
the reason why this method is here is so the user can override and provide their own algorithm. It still seems useful for the user to have some ability to override our default algorithm.

One of the reasons I didn't consolidate the two was because this method and the CTOR for Decoder are disjoint. I debated using a `Function` but that seemed like overkill and we are java6 anyways. I'm open to better suggestions.